contracts/peg/NonCustodialPSM.sol

```diff
@@ -0,0 +1,456 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {Decimal} from "../external/Decimal.sol";
+import {Constants} from "../Constants.sol";
+import {OracleRef} from "./../refs/OracleRef.sol";
+import {RateLimited} from "./../utils/RateLimited.sol";
+import {IPCVDeposit, PCVDeposit} from "./../pcv/PCVDeposit.sol";
+import {INonCustodialPSM} from "./INonCustodialPSM.sol";
+import {GlobalRateLimitedMinter} from "./../utils/GlobalRateLimitedMinter.sol";
+import {Math} from "@openzeppelin/contracts/utils/math/Math.sol";
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import {SafeCast} from "@openzeppelin/contracts/utils/math/SafeCast.sol";
+import {SafeERC20} from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import {ReentrancyGuard} from "@openzeppelin/contracts/security/ReentrancyGuard.sol";
+
+/// @notice Peg Stability Module that holds no funds.
+/// On a mint, it transfers all proceeds to a PCV Deposit and then deposits them into the deposit's target
+/// When funds are needed for a redemption, they are simply pulled from the PCV Deposit
+contract NonCustodialPSM is
+    INonCustodialPSM,
+    RateLimited,
+    OracleRef,
+    ReentrancyGuard
+{
+    using Decimal for Decimal.D256;
+    using SafeCast for *;
+    using SafeERC20 for IERC20;
+
+    /// @notice the fee in basis points for selling asset into FEI
+    uint256 public override mintFeeBasisPoints;
+
+    /// @notice the fee in basis points for buying the asset for FEI
+    uint256 public override redeemFeeBasisPoints;
+
+    /// @notice the PCV deposit target to deposit and withdraw from
+    IPCVDeposit public override pcvDeposit;
+
+    /// @notice the token this PSM will exchange for FEI
+    /// This token will be set to WETH9 if the bonding curve accepts eth
+    IERC20 public immutable override underlyingToken;
+
+    /// @notice Rate Limited Minter contract that will be called when FEI needs to be minted
+    GlobalRateLimitedMinter public immutable rateLimitedMinter;
```

Comment:
I don't see why this shouldn't be changeable by governance - it'll save us several contract deploys if/when we upgrade the minter contract.

Reply:
the minter contract should be immutable as changing it everywhere it is used is a massive pain

Reply:
That doesn't make any sense; redeploying every contract where the minter is used, because the minter immutable, is much more of a pain than calling a function on every contract where the minter is used to change it. 

Also, please don't resolve comments like this until we come to some kind of consensus.

Reply:
ok, now non immutable and changeable by governance